### PR TITLE
feat: ⚡️ add catalog to website

### PR DESCRIPTION
**Pull request by DistributedDoge**
EDIT: Working on #29, by making `catalog` notebook which can be re-run to extract schema from files in `tables` folder.

Updated preview => https://distributeddoge.github.io/gitcoin-grants-data-portal/catalog.html

-   `make publish` will now cause all notebook cells to re-run, before sending it to github pages.
- there is technically a detailed sidebar TOC you can preview by setting `toc-depth: 2` but then it covers dataframes when scrolling, so I think links are easier on the eyes.

### CODE REVIEW

File: portal/catalog.ipynb [hunk 1]
```diff
@@ -0,0 +1,2615 @@
...
+   "metadata": {},
+   "source": [
+    "---\n",
+    "title: \"Catalog\"\n",
+    "format:\n",
+    "    html:\n",
+    "        toc: true\n",
+    "        theme: darkly\n",
+    "        page-layout: full\n",
+    "        anchor-sections: true\n",
```
Comment (davidgasquez) [inline portal/catalog.ipynb]: We could force Quarto to run this notebook when rendering with this:

```
execute: 
  enabled: true
```

https://quarto.org/docs/tools/jupyter-lab.html#cell-execution

File: portal/catalog.ipynb [hunk 2]
```diff
@@ -0,0 +1,2615 @@
...
+     "output_type": "stream",
+     "text": [
+      "Processed Files:\n",
+      "application_answers.parquet\n",
+      "chain_metadata.parquet\n",
+      "passport_scores.parquet\n",
+      "projects.parquet\n",
+      "round_applications.parquet\n",
+      "round_votes.parquet\n",
+      "rounds.parquet\n",
```
Comment (davidgasquez) [inline portal/catalog.ipynb]: Could we print this in Markdown with links to their table headers?

**Review by davidgasquez — COMMENTED** [review]
Super quick review (haven't looked at the actual notebook code). 

Not needed for this iteration but I was wondering if there is a way to use the `sidebar` (showing all the tables) to make navigation easier.

Awesome work again @DistributedDoge! This is going to make data discovery muuuuch better.

File: portal/catalog.ipynb (hunk 2 above)
Comment (DistributedDoge) [reply portal/catalog.ipynb]: Good idea, will do.

File: portal/catalog.ipynb (hunk 1 above)
Comment (DistributedDoge) [reply portal/catalog.ipynb]: Done.

File: portal/catalog.ipynb (hunk 2 above)
Comment (DistributedDoge) [reply portal/catalog.ipynb]: Done. Two way links - clicking filename header will bring you back to file listing.

**Review by davidgasquez — APPROVED** [review]
Looks incredible @DistributedDoge!  

Ready to merge or is there anythihng else lingering you want to add here?

**Comment by DistributedDoge** [comment]
Ready to merge, let's just inspect the portal later to see if it actually worked.

**Comment by davidgasquez** [comment]
Seems the [CI worked](https://github.com/davidgasquez/gitcoin-grants-data-portal/actions/runs/7433555136/job/20226710138) but the page is not there yet. :mag: 

It should appear here: https://davidgasquez.github.io/gitcoin-grants-data-portal/catalog.html

**Comment by DistributedDoge** [comment]
@davidgasquez that is because CI run does not update github pages. 

Last update to github pages happened 2 weeks ago. I think one of your github-actions refactorings dropped that functionality from CI.

my workflow to update portal was to manually:

- make run
- make publish

**Comment by davidgasquez** [comment]
> I think one of your github-actions refactorings dropped that functionality from CI.

Yep! Need to get that back. Didn't meant to drop it! Probably a great opportunity to push it to IPFS too (and even `gitcoin-data.eth.limo`)! :blush:

**Pull request merged**